Source/MediaInfo/Audio/File_Usac.cpp

```diff
@@ -1743,11 +2178,25 @@ void File_Usac::UsacFrame(size_t BitsNotIncluded)
     {
         switch (C.usacElements[elemIdx].usacElementType&3)
         {
-            case ID_USAC_SCE                                    : UsacSingleChannelElement(); elemIdx=C.usacElements.size()-1; break;
-            case ID_USAC_CPE                                    : UsacChannelPairElement(); elemIdx=C.usacElements.size()-1; break;
-            case ID_USAC_LFE                                    : UsacLfeElement(); elemIdx=C.usacElements.size()-1; break;
-            case ID_USAC_EXT                                    : UsacExtElement(elemIdx, usacIndependencyFlag); break;
-            default                                             : ; //Not parsed
+            case ID_USAC_SCE:
+                UsacSingleChannelElement(usacIndependencyFlag);
+                if (C.coreSbrFrameLengthIndex>=coreSbrFrameLengthIndex_Mapping_Size || coreSbrFrameLengthIndex_Mapping[C.coreSbrFrameLengthIndex].sbrRatioIndex)
+                    elemIdx=C.usacElements.size()-1;
+                break;
+            case ID_USAC_CPE:
+                UsacChannelPairElement(usacIndependencyFlag);
+                if (C.coreSbrFrameLengthIndex>=coreSbrFrameLengthIndex_Mapping_Size || coreSbrFrameLengthIndex_Mapping[C.coreSbrFrameLengthIndex].sbrRatioIndex)

```

Comment:
Add:

```
   if (C.coreSbrFrameLengthIndex>=coreSbrFrameLengthIndex_Mapping_Size || coreSbrFrameLengthIndex_Mapping[C.coreSbrFrameLengthIndex].sbrRatioIndex || !Aac_sampling_frequency[C.sampling_frequency_index])
```

(for smooth stop of parsing in case of unknown frequency index)